internal/providers/terraform/aws/config_config_rule.go

```diff
@@ -0,0 +1,257 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetConfigRuleItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_config_config_rule",
+		RFunc: NewConfigRule,
+	}
+}
+
+func NewConfigRule(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	region := d.Get("region").String()
+
+	var monthlyConfigItems *decimal.Decimal
+	if u != nil && u.Get("monthly_config_items").Exists() {
+		monthlyConfigItems = decimalPtr(decimal.NewFromInt(u.Get("monthly_config_items").Int()))
+	}
+
+	var monthlyCustomConfigItems *decimal.Decimal
+	if u != nil && u.Get("monthly_custom_config_items").Exists() {
+		monthlyCustomConfigItems = decimalPtr(decimal.NewFromInt(u.Get("monthly_custom_config_items").Int()))
+	}
+
+	monthlyConfigRules := decimal.Zero
+	if u != nil && u.Get("monthly_rule_evaluations").Exists() {
+		monthlyConfigRules = decimal.NewFromInt(u.Get("monthly_rule_evaluations").Int())
+	}
+
+	monthlyConformancePacks := decimal.Zero
+	if u != nil && u.Get("monthly_conformance_pack_evaluations").Exists() {
+		monthlyConformancePacks = decimal.NewFromInt(u.Get("monthly_conformance_pack_evaluations").Int())
+	}
+
+	var rulesTiers = map[string]decimal.Decimal{
+		"tierOne":   decimal.Zero,
+		"tierTwo":   decimal.Zero,
+		"tierThree": decimal.Zero,
+	}
+
+	var packsTiers = map[string]decimal.Decimal{
+		"tierOne":   decimal.Zero,
+		"tierTwo":   decimal.Zero,
+		"tierThree": decimal.Zero,
+	}
+
+	configRulesTiers := []int64{100000, 400000}
+	conformancePacksTiers := []int64{1000000, 24000000}
+
+	configRulesQuantities := calculateTierItems(monthlyConfigRules, rulesTiers, configRulesTiers)
+	conformancePacksQuantities := calculateTierItems(monthlyConformancePacks, packsTiers, conformancePacksTiers)
+
+	rulesTierOne := configRulesQuantities["tierOne"]
+	rulesTierTwo := configRulesQuantities["tierTwo"]
+	rulesTierThree := configRulesQuantities["tierThree"]
+
+	packsTierOne := conformancePacksQuantities["tierOne"]
+	packsTierTwo := conformancePacksQuantities["tierTwo"]
+	packsTierThree := conformancePacksQuantities["tierThree"]
+
```

Comment:
Hi @ekifel maybe take a look at the below, there's a function you can call `usage.CalculateTierRequests` located here https://github.com/infracost/infracost/blob/417feb617a40ea7d033e1cd55aaab7d1f56c3ef0/internal/usage/tier_requests.go#L17 this will return you exactly what you need. You can remove a lot of code, & potentially create a function that returns the config cost component for tiered resources too.. https://github.com/infracost/infracost/blob/417feb617a40ea7d033e1cd55aaab7d1f56c3ef0/internal/providers/terraform/aws/acmpca_certificate_authority.go#L38-L60

Reply:
Hello, @robh007 ! Thanks a lot!
I didn't see this method when I write it.
But @alikhajeh1  said that CalculateTier method maybe will be refactored soon ( Issue #389 ) and I decided that I'll wait for it coz use-logic could change and it will need to refactor again

Reply:
@ekifel it might be a good idea to do the refactor mentioned in https://github.com/infracost/infracost/issues/389 inside this PR so we could test that it works ok with this resource and update the other resource using the function at the same time? it can be simpler to refactor when there are use-cases for it :)